src/Chapter3.hs

```diff
@@ -480,6 +514,88 @@ After defining the city, implement the following functions:
    and at least 10 living __people__ inside in all houses of the city totally.
 -}
 
+data UpToFour a
+  = One a
+  | Two a a
+  | Three a a a
+  | Four a a a a
+  deriving Show
+
+instance Foldable UpToFour where
+  foldMap f (One x           ) = f x
+  foldMap f (Two x1 x2       ) = f x2 <> foldMap f (One x1)
+  foldMap f (Three x1 x2 x3  ) = f x3 <> foldMap f (Two x1 x2)
+  foldMap f (Four x1 x2 x3 x4) = f x4 <> foldMap f (Three x1 x2 x3)
```

Comment:
That's a nice abstraction you have here! It's indeed very flexible and general :slightly_smiling_face: 
And it's very clever how you use to store values of type `Person` :clap: